wrench/src/main.rs

```diff
@@ -442,211 +425,133 @@ fn main() {
-
-                println!("-    {:7} {:7} {:7}", "10th", "avg", "90th");
-                println!(
-                    "ms   {:4.3} {:4.3} {:4.3}",
-                    val_10th_pct,
-                    avg_ms,
-                    val_90th_pct
-                );
-                println!(
-                    "fps  {:4.3} {:4.3} {:4.3}",
-                    1000. / val_10th_pct,
-                    1000. / avg_ms,
-                    1000. / val_90th_pct
-                );
-                break;
-            }
-        }
-
-        let event = match window {
-            WindowWrapper::Headless(..) => glutin::Event::Awakened,
-            WindowWrapper::Window(ref window, _) => window.wait_events().next().unwrap(),
-        };
+        let mut events = Vec::new();
 
-        if let Some(limit) = limit_seconds {
-            if (time::SteadyTime::now() - time_start) >= limit {
-                break 'outer;
+        match window {
+            WindowWrapper::Headless(..) => {
+                events.push(glutin::Event::Awakened);
+            }
+            WindowWrapper::Window(ref window, _) => {
+                events.push(window.wait_events().next().unwrap());
+                events.extend(window.poll_events());
             }
         }
 
-        match event {
-            glutin::Event::Awakened => {
-                let (width, height) = window.get_inner_size_pixels();
-                let dim = DeviceUintSize::new(width, height);
-                wrench.update(dim);
-
-                let frame_num = thing.do_frame(&mut wrench);
-                unsafe {
-                    CURRENT_FRAME_NUMBER = frame_num;
-                }
+        let mut do_frame = false;
 
-                if show_help {
-                    wrench.show_onscreen_help();
+        for event in events {
+            match event {
+                glutin::Event::Closed => {
+                    break 'outer;
                 }
 
-                wrench.render();
-                window.swap_buffers();
+                glutin::Event::MouseWheel(delta, _, _) => {
+                    let dy = match delta {
+                        glutin::MouseScrollDelta::LineDelta(_, dy) => dy,
+                        glutin::MouseScrollDelta::PixelDelta(_, dy) => dy,
+                    };
 
-                let now = time::SteadyTime::now();
-                let dur = now - last;
+                    let delta = if dy > 0.0 {
+                        1.1
+                    } else {
+                        0.9
+                    };
 
-                min_time = min(min_time, dur);
-                min_min_time = min(min_min_time, dur);
-                max_time = max(max_time, dur);
-                max_max_time = max(max_max_time, dur);
-                sum_time = sum_time + dur;
+                    let current_zoom = wrench.get_page_zoom();
+                    let new_zoom_factor = ZoomFactor::new(current_zoom.get() * delta);
```

Comment:
[thinking out loud] this is a bit weird since (zoom in + zoom out) isn't an identity function? Should we instead just do addition, with a min/max zoom level?

Reply:
Fixed